### PR TITLE
remove crosby, protbuf dependency & sonarbug

### PR DESCRIPTION
**Pull request by rtroilo**
This PR removes the crosby.binary osm pbf implementation from our repository and replace them with a dependency to: 
```xml
    <dependency>
        <groupId>org.openstreetmap.pbf</groupId>
        <artifactId>osmpbf</artifactId>
        <version>1.5.0</version>
    </dependency>
 ```
It also removes the dependency  to google protobuf from the `oshdb` module. Because we only used the varint encoding from that library and this encoding is a standard, so we provide our own implementation for that.

### Corresponding issue
Closes #269 

### New or changed dependencies
- removed: com.google.protobuf.protobuf-java from oshdb module
- added: org.openstreetmap.pbf.osmpbf to oshdb-tool/oshdb-parser module

### Checklist
- [x] My code follows the [code-style](https://github.com/GIScience/oshdb/blob/master/CONTRIBUTING.md) rules, and I have checked on the [static analyses](https://jenkins.ohsome.org/job/oshdb/view/change-requests/) and [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) (if applicable) results
- [x] I have commented my code
- [x] I have written javadoc (required for public classes and methods)
- [x] I have added sufficient unit tests
- ~I have made corresponding changes to the [documentation](https://github.com/GIScience/oshdb/tree/master/documentation)~
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/oshdb/blob/master/CHANGELOG.md)
- ~I have adjusted the [examples](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-examples) or [created an issue](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-examples/-/issues/new) in the corresponding repository~
- ~I have adjusted the [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) or [created an issue](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-benchmarks/-/issues/new) in the corresponding repository~

### CODE REVIEW

**Comment by tyrasd** [comment]
> * [ ] I have updated the CHANGELOG.md

I'd say this is not necessary in this case, since the change is purely internal and doesn't change anything for end users. I'll strike it out above.

File: oshdb-tool/oshpbf-parser/pom.xml [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
 
-		<dependency>
-			<groupId>com.google.protobuf</groupId>
-			<artifactId>protobuf-java</artifactId>
-			<version>${protobuf.version}</version>
-		</dependency>
+        <dependency>
+            <groupId>org.openstreetmap.pbf</groupId>
+            <artifactId>osmpbf</artifactId>
+            <version>1.5.0</version>
```
Comment (tyrasd) [inline oshdb-tool/oshpbf-parser/pom.xml:26]: can you please introduce a property (e.g. `${osmpbf.version}`) for this?

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHNodeImpl.java [hunk 2]
```diff
@@ -18,6 +17,10 @@
 import org.heigit.ohsome.oshdb.util.bytearray.ByteArrayOutputWrapper;
 import org.heigit.ohsome.oshdb.util.bytearray.ByteArrayWrapper;
 
+/**
+ * An implementation of the {@link OSHNode} interface.
+ *
```
Comment (tyrasd) [inline oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHNodeImpl.java:22]: maybe skip this "empty" line?

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHRelationImpl.java [hunk 3]
```diff
@@ -30,6 +30,10 @@
 import org.heigit.ohsome.oshdb.util.bytearray.ByteArrayOutputWrapper;
 import org.heigit.ohsome.oshdb.util.bytearray.ByteArrayWrapper;
 
+/**
+ * An implementation of the {@link OSHRelation} interface.
+ *
```
Comment (tyrasd) [inline oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHRelationImpl.java:35]: skip empty line?

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHWayImpl.java [hunk 4]
```diff
@@ -27,6 +27,10 @@
 import org.heigit.ohsome.oshdb.util.bytearray.ByteArrayOutputWrapper;
 import org.heigit.ohsome.oshdb.util.bytearray.ByteArrayWrapper;
 
+/**
+ * An implementation of the {@link OSHWay} interface.
+ *
```
Comment (tyrasd) [inline oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHWayImpl.java:32]: skip empty line?

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayOutputWrapper.java [hunk 5]
```diff
@@ -1,37 +1,61 @@
 package org.heigit.ohsome.oshdb.util.bytearray;
 
-import com.google.protobuf.CodedOutputStream;
 import java.io.IOException;
 
+/**
+ * A wrapper around {@link OSHDBByteArrayOutputStream} for extending varint encoding methods.
+ *
```
Comment (tyrasd) [inline oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayOutputWrapper.java:7]: skip empty line?

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayWrapper.java [hunk 6]
```diff
@@ -1,18 +1,13 @@
 package org.heigit.ohsome.oshdb.util.bytearray;
 
-import java.io.IOException;
 import java.util.Arrays;
 
+/**
+ * An wrapper class around an byte array, offering various read methods.
+ *
```
Comment (tyrasd) [inline oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayWrapper.java:7]: skip empty line?

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayWrapper.java [hunk 7]
```diff
@@ -62,163 +57,70 @@ public int hasLeft() {
...
-  public long readRawVarint64() throws IOException {
-    // Implementation notes:
-    // Optimized for one-byte values, expected to be common.
-    // The particular code below was selected from various candidates
-    // empirically, by winning VarintBenchmark.
-    // Sign extension of (signed) Java bytes is usually a nuisance, but
-    // we exploit it here to more easily obtain the sign of bytes read.
-    // Instead of cleaning up the sign extension bits by masking eagerly,
-    // we delay until we find the final (positive) byte, when we clear all
-    // accumulated bits with one xor. We depend on javac to constant fold.
```
Comment (tyrasd) [inline oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayWrapper.java:154]: is this "fast path" not needed anymore?

**Review by tyrasd — COMMENTED** [review]
just a few comments inline

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayWrapper.java (hunk 7 above)
Comment (rtroilo) [reply oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayWrapper.java:154]: Actually not, this implementation was originally taken from the google protobuf library. I also looked into other implementations like SerializationUtils from apache orc or BytesUtils from apache parquet both use a more simpler and more maintainable version of varint decoding. so I opt to do the optimization for one-byte values, like protobuf but then switch to the simpler implementation like orc or parquet.

Comment (tyrasd) [reply oshdb/src/main/java/org/heigit/ohsome/oshdb/util/bytearray/ByteArrayWrapper.java:154]: :+1: thanks for the explanation.

**Review by tyrasd — APPROVED** [review]

**Review by tyrasd — APPROVED** [review]

**Review by tyrasd — APPROVED** [review]

**Pull request merged**